src/content/github/file/main.tsx

```diff
@@ -151,34 +187,36 @@ async function process(metadata: FileMetadata): Promise<void> {
       previousElement: globals.coverageButton!,
       onClick: handleComponentClick,
       selectedOptions: selectedComponents,
-    })
-      .then(({ button, list }) => {
-        globals.componentsButton = button;
-        globals.componentsDrop = new Drop({
-          target: button,
-          content: list,
-          classes: "drop-theme-arrows codecov-z1 codecov-bg-white",
-          position: "bottom right",
-          openOn: "click",
-        });
-      })
+    }).then(({ button, list }) => {
+      globals.componentsButton = button;
+      globals.componentsDrop = new Drop({
+        target: button,
+        content: list,
+        classes: "drop-theme-arrows codecov-z1 codecov-bg-white",
+        position: "bottom right",
+        openOn: "click",
+      });
+    });
   }
 
+  // If commit sha is defined use that, otherwise just branch name
+  const getReportFn = metadata.commit ? getCommitReport : getBranchReport;
+
   let coverageReportResponses: Array<FileCoverageReportResponse>;
   try {
     if (selectedFlags?.length > 0) {
       coverageReportResponses = await Promise.all(
-        selectedFlags.map((flag) => getCommitReport(metadata, flag, undefined))
+        selectedFlags.map((flag) => getReportFn(metadata, flag, undefined))
       );
     } else if (selectedComponents?.length > 0) {
       coverageReportResponses = await Promise.all(
         selectedComponents.map((component) =>
-          getCommitReport(metadata, undefined, component)
+          getReportFn(metadata, undefined, component)
         )
       );
     } else {
       coverageReportResponses = await Promise.all([
-        await getCommitReport(metadata, undefined, undefined),
+        await getReportFn(metadata, undefined, undefined),
       ]);
```

Comment:
Not 100% certain on what's going on here, do we need this `Promise.all` anymore?

Second question, does `getCommitReport` itself return a promise, If not can we remove the `await` in front of `getReportFn`?

Reply:
We don't, I think the original author did this to keep it as an array, but I suppose `[await getReportFn()]` is better.

Second Q yes.

Reply:
Sorry, slightly confused. Does `getReportFn` return a chained promise (i.e. a promise that when resolved returns another promise)?

Reply:
No, it doesn't return a chained promise. It just resolves to the report results.

Reply:
Ahh ok ok, the previous setup was confusing me around that, good to know 👍